### PR TITLE
Add Jenkins setup for running nightly tests on examples

### PR DESCRIPTION
**Pull request by kalyc**
## Description ##
Add Jenkins setup for running nightly tests on examples

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Add Jenkins setup for running nightly tests on examples for python2 & python3 on GPU

## Comments ##
- PR as per discussion offline with @marcoabreu

### CODE REVIEW

**Comment by kalyc** [comment]
@marcoabreu @ankkhedia

**Comment by kalyc** [comment]
@mxnet-label-bot add [pr-awaiting-review]

File: ci/docker/runtime_functions.sh [hunk 1]
```diff
@@ -1125,6 +1125,30 @@ nightly_tutorial_test_ubuntu_python2_gpu() {
...
+nightly_example_test_ubuntu_python3_gpu() {
+    set -ex
+    cd /work/mxnet/docs
+    export BUILD_VER=tutorial
+    export MXNET_DOCS_BUILD_MXNET=0
+    make html
+    export MXNET_STORAGE_FALLBACK_LOG_VERBOSE=0
+    export MXNET_TEST_KERNEL=python3
+    cd /work/mxnet/tests/examples
+    nosetests-3.4 --with-xunit --xunit-file nosetests_tutorials.xml test_examples.py --nologcapture
```
Comment (roywei) [inline ci/docker/runtime_functions.sh]: Does this PR depend on https://github.com/apache/incubator-mxnet/pull/13271 to be merged?

File: ci/docker/runtime_functions.sh [hunk 2]
```diff
@@ -1125,6 +1125,30 @@ nightly_tutorial_test_ubuntu_python2_gpu() {
     nosetests-3.4 --with-xunit --xunit-file nosetests_tutorials.xml test_tutorials.py --nologcapture
 }
 
+nightly_example_test_ubuntu_python3_gpu() {
```
Comment (marcoabreu) [inline ci/docker/runtime_functions.sh]: Please review everything in here. It seems like this has just been copy and pasted

File: ci/docker/runtime_functions.sh [hunk 3]
```diff
@@ -1125,6 +1125,30 @@ nightly_tutorial_test_ubuntu_python2_gpu() {
...
+nightly_example_test_ubuntu_python2_gpu() {
+    set -ex
+    cd /work/mxnet/docs
+    export BUILD_VER=tutorial
+    export MXNET_DOCS_BUILD_MXNET=0
+    make html
+    export MXNET_STORAGE_FALLBACK_LOG_VERBOSE=0
+    export MXNET_TEST_KERNEL=python2
+    cd /work/mxnet/tests/examples
+    nosetests-2.7 --with-xunit --xunit-file nosetests_tutorials.xml test_examples.py --nologcapture
```
Comment (ankkhedia) [inline ci/docker/runtime_functions.sh]: This should be examples and not tutorials

File: ci/docker/runtime_functions.sh (hunk 1 above)
Comment (anirudhacharya) [reply ci/docker/runtime_functions.sh]: as mentioned here - https://github.com/apache/incubator-mxnet/pull/13271#discussion_r233745860, please combine the PRs

**Comment by marcoabreu** [comment]
@jlcontreras

File: ci/docker/runtime_functions.sh [hunk 4]
```diff
@@ -1125,6 +1125,30 @@ nightly_tutorial_test_ubuntu_python2_gpu() {
...
+    export MXNET_DOCS_BUILD_MXNET=0
+    make html
+    export MXNET_STORAGE_FALLBACK_LOG_VERBOSE=0
+    export MXNET_TEST_KERNEL=python3
+    cd /work/mxnet/tests/examples
+    nosetests-3.4 --with-xunit --xunit-file nosetests_tutorials.xml test_examples.py --nologcapture
+
+}
+
+nightly_example_test_ubuntu_python2_gpu() {
```
[the remark this answers is not in the document]
Comment (kalyc) [reply ci/docker/runtime_functions.sh]: We haven't moved off of python2 yet - the other nightly tests also run for python2

File: ci/docker/runtime_functions.sh [hunk 5]
```diff
@@ -1125,6 +1125,30 @@ nightly_tutorial_test_ubuntu_python2_gpu() {
     nosetests-3.4 --with-xunit --xunit-file nosetests_tutorials.xml test_tutorials.py --nologcapture
 }
 
+nightly_example_test_ubuntu_python3_gpu() {
+    set -ex
+    cd /work/mxnet/docs
+    export BUILD_VER=tutorial
+    export MXNET_DOCS_BUILD_MXNET=0
+    make html
```
[the remark this answers is not in the document]
Comment (kalyc) [reply ci/docker/runtime_functions.sh]: Yes missed on my end, removed.

**Comment by kalyc** [comment]
Addressed comments and removed extraneous commands from jenkins setup

**Comment by stu1130** [comment]
@marcoabreu @anirudhacharya @ankkhedia looks like your comments were addressed?

**Comment by marcoabreu** [comment]
Did we test this yet?

**Comment by kalyc** [comment]
@marcoabreu we have cut a TT at your end for assistance in helping with testing this setup. Could you help us out with it?

**Comment by kalyc** [comment]
@Chancebair could you help with testing this PR?

**Comment by nswamy** [comment]
@kalyc could you please follow up and take it to completion.

**Comment by Roshrini** [comment]
@kalyc Thanks for working on this! Is this PR good to go? Did it pass tests on jenkins setup?

**Comment by kalyc** [comment]
@Roshrini looks like there is a build failure with this setup - http://jenkins.mxnet-ci-dev.amazon-ml.com/job/test-kalyc-NightlyTestsForBinaries/job/jenkins_setup_testing_examples/

Requesting @ankkhedia @roywei to take a look. Please let me know if this PR needs to be closed accordingly.

**Comment by kalyc** [comment]
Error log - 

======================================================================
FAIL: test_examples.test_cifar_gpu
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/work/mxnet/tests/examples/test_examples.py", line 79, in test_cifar_gpu
    ['python', os.path.join(example_dir, 'train_cifar10.py'), '--num-epochs', '5', 'gpus', '0'])
AssertionError

----------------------------------------------------------------------
Ran 2 tests in 6024.068s

**Comment by kalyc** [comment]
As discussed offline @ankkhedia and @roywei will take this PR forward. Closing this PR for now.

**Pull request closed without merging**